src/main/java/uk/nhs/hee/trainee/details/service/ProgrammeMembershipService.java

```diff
@@ -62,11 +65,32 @@ public Optional<ProgrammeMembership> updateProgrammeMembershipForTrainee(String
         || programmeMembership.getConditionsOfJoining().signedAt() == null) {
 
       // Restore the Conditions of Joining if it exists after this PM was previously deleted.
-      for (String id : programmeMembership.getTisId().split(",")) {
-        Optional<ConditionsOfJoining> conditionsOfJoining = cachingDelegate.getConditionsOfJoining(
-            id);
-        conditionsOfJoining.ifPresent(programmeMembership::setConditionsOfJoining);
-        // All results should be the same, but iterating through all IDs ensures a clean cache.
+      // FIXME: push all PMs through tis-trainee-sync to flush 2 and 3 so they can be removed?
+      // 3 scenarios:
+      try {
+        //1. new uuid PM, with CoJ also cached against new uuid *THE FUTURE*
+        UUID uuid = UUID.fromString(programmeMembership.getTisId());
+        Optional<ConditionsOfJoining> conditionsOfJoiningUuid
+            = cachingDelegate.getConditionsOfJoining(uuid.toString());
+        if (conditionsOfJoiningUuid.isPresent()) {
+          programmeMembership.setConditionsOfJoining(conditionsOfJoiningUuid.get());
+        } else {
+          //2. new uuid PM, with CoJ cached against old delimited cm ids *THE PRESENT*
+          for (Curriculum curriculum : programmeMembership.getCurricula()) {
+            Optional<ConditionsOfJoining> conditionsOfJoiningId
+                = cachingDelegate.getConditionsOfJoining(curriculum.getCurriculumTisId());
```

Comment:
We need the curriculum tisId here, not the curriculumTisId. Which might mean another PR for trainee-sync needs to be merged as well.